docs/sources/mimir/configure/about-versioning/index.md

```diff
@@ -0,0 +1,913 @@
+```
+
+### compactor
+
+The `compactor` block configures the compactor component.
+
+```yaml
+# (experimental) If enabled, will delete the bucket-index, markers and debug
+# files in the tenant bucket when there are no blocks left in the index.
+# CLI flag: -compactor.no-blocks-file-cleanup-enabled
+[no_blocks_file_cleanup_enabled: <boolean> | default = false]
+```
+
+### memcached
+
+The `memcached` block configures the Memcached-based caching backend. The supported CLI flags `<prefix>` used to reference this configuration block are:
+
+- `blocks-storage.bucket-store.chunks-cache`
+- `blocks-storage.bucket-store.index-cache`
+- `blocks-storage.bucket-store.metadata-cache`
+- `query-frontend.results-cache`
+- `ruler-storage.cache`
+
+&nbsp;
+
+```yaml
+# (experimental) The size of the write buffer (in bytes). The buffer is
+# allocated for each connection to memcached.
+# CLI flag: -<prefix>.memcached.write-buffer-size-bytes
+[write_buffer_size_bytes: <int> | default = 4096]
+
+# (experimental) The size of the read buffer (in bytes). The buffer is allocated
+# for each connection to memcached.
+# CLI flag: -<prefix>.memcached.read-buffer-size-bytes
+[read_buffer_size_bytes: <int> | default = 4096]
+```
+
+### s3_storage_backend
+
+The s3_backend block configures the connection to Amazon S3 object storage backend. The supported CLI flags `<prefix>` used to reference this configuration block are:
+
+- `alertmanager-storage`
+- `blocks-storage`
+- `common.storage`
+- `ruler-storage`
+
+&nbsp;
+
+```yaml
+# (experimental) The S3 storage class to use, not set by default. Details can be
+# found at https://aws.amazon.com/s3/storage-classes/. Supported values are:
+# STANDARD, REDUCED_REDUNDANCY, GLACIER, STANDARD_IA, ONEZONE_IA,
+# INTELLIGENT_TIERING, DEEP_ARCHIVE, OUTPOSTS, GLACIER_IR, SNOW, EXPRESS_ONEZONE
+# CLI flag: -<prefix>.s3.storage-class
+[storage_class: <string> | default = ""]
+
+# (experimental) If enabled, it will use the default authentication methods of
+# the AWS SDK for go based on known environment variables and known AWS config
+# files.
+# CLI flag: -<prefix>.s3.native-aws-auth-enabled
+[native_aws_auth_enabled: <boolean> | default = false]
+
+# (experimental) The minimum file size in bytes used for multipart uploads. If
+# 0, the value is optimally computed for each object.
+# CLI flag: -<prefix>.s3.part-size
+[part_size: <int> | default = 0]
+
+# (experimental) If enabled, a Content-MD5 header is sent with S3 Put Object
+# requests. Consumes more resources to compute the MD5, but may improve
+# compatibility with object storage services that do not support checksums.
+# CLI flag: -<prefix>.s3.send-content-md5
+[send_content_md5: <boolean> | default = false]
+```
+
+## Deprecated features
+
+Deprecated features are usable up until the release that indicates their removal.
+
+The following features or configuration parameters are currently deprecated and will be **removed in Mimir 2.11**:
```

Comment:
Rather than having the reference to Mimir 2.11 in the docs template, what if we attached extra information to deprecated flags that specify when they were deprecated and when they'll be removed? Then we could generate this list without needing to change the template each release, and we can also easily check for any deprecated flags that should be removed in the next release.

I'm imagining something like this:

```go
DeprecatedIndexHeaderLazyLoadingIdleTimeout time.Duration `yaml:"index_header_lazy_loading_idle_timeout" category:"deprecated,since=2.10,remove_in=2.12"`
```

Reply:
`since` is ambiguous to mean either `as_of` or `after`. Therefore, I recommend using `as_of` instead of `since`.

Reply:
`remove_in` is an imperative to do something in the future. Is the intention to write this from today onward, or will this mostly be viewed from the future back in time? I recommend `removed_in`.

Reply:
> `remove_in` is an imperative to do something in the future. Is the intention to write this from today onward, or will this mostly be viewed from the future back in time? I recommend `removed_in`, where one could read this as (to be) `removed_in`.

My intention with the `remove_in` / `removed_in` is to codify when we plan to remove the flag, so we'd be talking about future plans. (We usually deprecate a flag in release X, and remove it altogether in release X+2.)

So perhaps something like `expected_to_be_removed_in` or `planned_removed_in`? What do you think @osg-grafana?

Reply:
> > `remove_in` is an imperative to do something in the future. Is the intention to write this from today onward, or will this mostly be viewed from the future back in time? I recommend `removed_in`, where one could read this as (to be) `removed_in`.
> 
> My intention with the `remove_in` / `removed_in` is to codify when we plan to remove the flag, so we'd be talking about future plans. (We usually deprecate a flag in release X, and remove it altogether in release X+2.)
> 
> So perhaps something like `expected_to_be_removed_in` or `planned_removed_in`? What do you think @osg-grafana?

Thank you for the clarification. Your intention matches my interpretation, so I recommend keeping as it is.